### PR TITLE
Position_list e2e tests

### PR DESCRIPTION
**Pull request by DevRozaDev**
\+ fix `positionsCounter` not decreasing when position is removed
\+ fix protocol trying to create `positionsCounter` for the caller if all his positions got removed beforehand
\+ `transferVerifyPosition`  snippet for position transferring
\+ `transferPosition` testUtil function

### CODE REVIEW

File: test/position_list.test.ts [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+        })
+      }
+    }
+    // 3 more with different tick settings but overlapping
+    {
+      const tickIndexes: Array<[bigint, [bigint, bigint]]> = [
+        [-10n, 10n],
+        [-20n, -10n],
+        [10n, 20n]
+      ].map(([i1, i2], index) => {
```
Comment (Sniezka1927) [inline test/position_list.test.ts:145]: There is no need to map array, you can access the index and value in for loop:
```ts
const tickIndexes = [[-10n, 10n],[-20n, -10n],[10n, 20n]]
for (const [index, [lowerTick, upperTick]] of tickIndexes.entries()) {
    ...
}
```

File: test/position_list.test.ts [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+
+    const { sqrtPrice: slippageLimitLower } = await getPool(invariant, poolKey)
+
+    const minMaxTicks: Array<[number, number]> = [
+      [0, 1],
+      [0, 1],
+      [0, 2],
+      [1, 4]
+    ]
+    for (let minMaxTick of minMaxTicks) {
```
Comment (Sniezka1927) [inline test/position_list.test.ts:257]: Here you can use as well array destructuring
```ts
for (const [minIndex, maxIndex] of minMaxTick) {
    const minTick = tickIndexes[minIndex]
    const maxTick = tickIndexes[maxIndex]
    ...
}
```

File: test/position_list.test.ts [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+      )
+    }
+  })
+  test('add multiple positions', async () => {
+    const feeTier = await newFeeTier(getBasicFeeTickSpacing()[0], 3n)
+    const poolKey = await newPoolKey(tokenX.contractId, tokenY.contractId, feeTier)
+    const { sqrtPrice: slippageLimitLower } = await getPool(invariant, poolKey)
+
+    // remove middle position
+    {
```
Comment (Sniezka1927) [inline test/position_list.test.ts:281]: You should check also position list length after each removal

File: src/snippets.ts [hunk 4]
```diff
@@ -131,3 +134,33 @@ export const initBasicSwap = async (
   }
   expect(poolAfter).toMatchObject(poolExpected)
 }
+
+export const transferVerifyPosition = async (
+  invariant: InvariantInstance,
+  sender: PrivateKeyWallet,
```
Comment (Sniezka1927) [inline src/snippets.ts:140]: owner would be more clear

File: src/snippets.ts [hunk 5]
```diff
@@ -131,3 +134,33 @@ export const initBasicSwap = async (
...
+
+  const { owner: previousOwner, ...toTransfer } = await getPosition(
+    invariant,
+    sender.address,
+    index
+  )
+  const senderLastPositionBefore = await getPosition(invariant, sender.address, senderListLength)
+  await transferPosition(invariant, sender, index, recipient)
+  const transferredPosition = await getPosition(invariant, recipient, recipientListLength + 1n)
+  const senderFirstPositionAfter = await getPosition(invariant, sender.address, index)
```
Comment (Sniezka1927) [inline src/snippets.ts:158]: It is not always first position

**Review by Sniezka1927 — CHANGES_REQUESTED** [review]

**Review by Sniezka1927 — CHANGES_REQUESTED** [review]

File: src/snippets.ts [hunk 6]
```diff
@@ -131,3 +134,29 @@ export const initBasicSwap = async (
   }
   expect(poolAfter).toMatchObject(poolExpected)
 }
+
+export const transferVerifyPosition = async (
```
Comment (wojciech-cichocki) [inline src/snippets.ts:138]: Maybe `transferAndVerifyPosition`

**Review by wojciech-cichocki — APPROVED** [review]

**Pull request merged**